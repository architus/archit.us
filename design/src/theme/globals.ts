import { css } from "linaria";

import { injectColorGlobals, color } from "./color";
import { injectMixinGlobals } from "./mixins";
import { injectTypographyGlobals } from "./typography";

// From https://hankchizljaw.com/wrote/a-modern-css-reset/
export const globals = css`
  :global() {
    ${injectColorGlobals()}
    ${injectTypographyGlobals()}
    ${injectMixinGlobals()}

    // Box sizing rules
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    // Remove default padding
    ul[class],
    ol[class] {
      padding: 0;
    }

    // Remove default margin
    body,
    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    p,
    ul,
    ol,
    li,
    figure,
    figcaption,
    blockquote,
    dl,
    dd {
      margin: 0;
    }

    // Set core body defaults
    body {
      margin: 0;
      min-height: 100vh;
      scroll-behavior: smooth;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      text-rendering: optimizeSpeed;

      line-height: 1.5;
      color: ${color("text")};
      background-color: ${color("bg")};
    }

    // Remove list styles on ul, ol elements with a class attribute
    ul[class],
    ol[class] {
      list-style: none;
    }

    // Inherit fonts for inputs and buttons
    input,
    button,
    textarea,
    select {
      font: inherit;
    }

    // Remove all animations and transitions for people that prefer not to see them
    @media (prefers-reduced-motion: reduce) {
      * {
        animation-duration: 0.01ms !important;
        animation-iteration-count: 1 !important;
        transition-duration: 0.01ms !important;
        scroll-behavior: auto !important;
      }
    }
  }
`;
