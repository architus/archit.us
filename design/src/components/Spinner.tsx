import { styled } from "linaria/react";
import React from "react";

import { color, Variant } from "@design/theme/color";
import { animation } from "@design/theme/motion";

const Styled = {
  Spinner: styled.svg<{ variant: Variant; size: string }>`
    --size: ${(props): string => props.size};
    --stroke: calc(var(--size) / 12);
    --color: ${(props): string => color(props.variant)};

    ${animation("rotate")}
    animation: rotate 2s linear infinite;

    z-index: 2;
    position: absolute;
    top: 50%;
    left: 50%;
    margin: calc(var(--size) / -2) 0 0 calc(var(--size) / -2);
    width: var(--size);
    height: var(--size);
  `,
  Path: styled.circle`
    fill: none;
    stroke: var(--color);
    stroke-linecap: round;
    stroke-width: var(--stroke);

    ${animation("spinnerDash")}
    animation: spinnerDash 1.5s ease-in-out infinite;
  `,
};

type SpinnerProps = {
  variant?: Variant;
  size?: string;
  className?: string;
  style?: React.CSSProperties;
};

/**
 * Bootstrap-style spinner component
 */
const Spinner: React.FC<SpinnerProps> = ({
  variant = "primary",
  size = "64px",
  className,
  style,
}) => (
  <Styled.Spinner
    viewBox="0 0 50 50"
    variant={variant}
    size={size}
    className={className}
    style={style}
  >
    <Styled.Path cx="25" cy="25" r="20" />
  </Styled.Spinner>
);

export default Spinner;
