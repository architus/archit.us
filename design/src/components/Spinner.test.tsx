import { describe, it, expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";

import Spinner from "./Spinner";

describe("<Spinner />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Spinner
          variant="primary"
          size="64px"
          className="additional-class"
          style={{ backgroundColor: "blue" }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
