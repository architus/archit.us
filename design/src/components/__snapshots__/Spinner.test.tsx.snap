// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<svg
  className="additional-class Spinner_sjge0pw"
  size="64px"
  style={
    Object {
      "--sjge0pw-0": "64px",
      "--sjge0pw-1": "var(--c-primary)",
      "backgroundColor": "blue",
    }
  }
  viewBox="0 0 50 50"
>
  <circle
    className="Path_piekwga"
    cx="25"
    cy="25"
    r="20"
  />
</svg>
`;
